src/DimoCredit.sol

```diff
@@ -0,0 +1,209 @@
+ * @custom:coauthor Lorran Sutter (@LorranSutter)
+ * @custom:coauthor Dylan Moreland (@elffjs)
+ * @custom:coauthor Yevgeny Khessin (@zer0stars)
+ * @custom:coauthor Rob Solomon (@robmsolomon)
+ * @custom:contributor Allyson English (@aesdfghjkl666)
+ * 
+ * @dev non-transferable, ipso facto no approve logic
+ * @notice approve this contract on $DIMO token before minting
+ *         address is 0xE261D618a959aFfFd53168Cd07D12E37B26761db
+ *         1 DC == $0.001 USD
+ */
+contract DimoCredit is AccessControl {
+
+    /*//////////////////////////////////////////////////////////////
+                             Access Controls
+    //////////////////////////////////////////////////////////////*/
+    bytes32 public constant BURNER_ROLE = keccak256("BURNER_ROLE");
+    bytes32 public constant DC_ADMIN_ROLE = keccak256("DC_ADMIN_ROLE");
+
+    address public _receiver; ///@notice receives proceeds from sale of license
+
+    IDimoToken public _dimo;
+    NormalizedPriceProvider public _provider;
+    uint256 public _periodValidity;
+
+    uint256 constant public SCALING_FACTOR = 1 ether;
+
+    uint256 public _dimoCreditRate;
+
+    function receiver() external view returns (address receiver_) {
+        receiver_ = _receiver;
+    }
+
+    function dimoCreditRate() external view returns (uint256 dimoCreditRate_) {
+        dimoCreditRate_ = _dimoCreditRate;
+    }
+
+    function setDimoCreditRate(uint256 dimoCreditRate_) external onlyRole(DC_ADMIN_ROLE) {
+        _dimoCreditRate = dimoCreditRate_;
+        emit UpdateDimoCreditRate(_dimoCreditRate);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            Error Messages
+    //////////////////////////////////////////////////////////////*/
+    string INVALID_OPERATION = "DimoCredit: invalid operation";
+
+    /*//////////////////////////////////////////////////////////////
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+    event UpdateDimoCreditRate(uint256 rate);
+    ///@dev only used in mint and burn, not transferable
+    event Transfer(address indexed from, address indexed to, uint256 amount);
+
+    /*//////////////////////////////////////////////////////////////
+                            METADATA STORAGE
+    //////////////////////////////////////////////////////////////*/
+    string public name;
+
+    string public symbol;
+
+    uint8 public immutable decimals;
+
+    /*//////////////////////////////////////////////////////////////
+                              ERC20 STORAGE
+    //////////////////////////////////////////////////////////////*/
+    uint256 public totalSupply;
+
+    mapping(address => uint256) public balanceOf;
+    
+    /**
+     */
+    constructor(address receiver_, address provider_) {
+        
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+
+        _dimo = IDimoToken(0xE261D618a959aFfFd53168Cd07D12E37B26761db);
```

Comment:
should we pass this rather than hardcode?

Reply:
We learned with other contracts that is wise to do it just in case.